clients/cobol-lsp-vscode-extension/src/commands/SmartTabCommand.ts

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2020 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Broadcom, Inc. - initial API and implementation
+ */
+
+import * as vscode from "vscode";
+import { ProfileService } from "../services/ProfileService";
+
+const COBOL_SECTIONS_DEFINITION: number[] = [7, 11, 72, 80];
+
+/**
+ * Executes a smart tab command
+ */
+export function executeSmartTab() {
+    const editor = vscode.window.activeTextEditor;
+    if (editor == null) {
+        return;
+    }
+
+    if (editor.selection.isEmpty) {
+        const position = editor.selection.active;
+        const textLine = editor.document.lineAt(position.line);
+        const lineSize = textLine.text.length;
+
+        const nextPosition = getNextPosition(editor, position.character);
+        if (nextPosition > lineSize) {
+            navigateTo(editor, position.line, lineSize);
+            insertText(x => generateSpaces(nextPosition - lineSize)); 
+        }
+        navigateTo(editor, position.line, nextPosition);
+    }
+}
+
+/**
+ * Moves cursor of the editor to a needed position
```

Comment:
The comment body should start with a verb in the first form

Reply:
Fixed